src/include/migraphx/op/slice.hpp

```diff
@@ -98,10 +100,30 @@ struct slice
 
     shape normalize_compute_shape(std::vector<shape> inputs) const
     {
-        auto input_shape        = inputs[0];
-        auto t                  = input_shape.type();
-        const auto& old_lens    = input_shape.lens();
-        const auto& old_strides = input_shape.strides();
+        auto input_shape = inputs[0];
+        auto t           = input_shape.type();
+        // For a static shape, old_lens will be adjusted to a new size
+        // for those axes that are sliced.
+        // For dynamic shape, the adjusted old_lens become the new max values,
+        // while retaining the old mins and opts.
+        std::vector<std::size_t> new_mins;
+        std::vector<std::size_t> new_opts;
+        std::vector<std::size_t> old_lens;
+        std::vector<std::size_t> old_strides;
+        if(input_shape.dynamic())
+        {
+            old_lens = input_shape.max_lens();
+            new_mins = input_shape.min_lens();
+            new_opts = input_shape.opt_lens();
+        }
+        else
+        {
+            old_lens = input_shape.lens();
+            // For static shape (including eval after a dynamic input) the strides are indexed into
+            // the pre-slice array, so they are larger than the apparent size of the resulting
+            // shape.
+            old_strides = input_shape.strides();
+        }
 
         if(std::any_of(
                axes.begin(), axes.end(), [&](auto i) { return (i >= old_lens.size() and i < 0); }))
```

Comment:
done

Reply:
I don't see change here as well.

Reply:
I found a bigger problem when I tried this:  the test I wrote for an (invalid) negative axis doesn't work because the negative value wasn't passed along to the Slice operator.  I added debug code to print the axes and a -1 test input gets converted to 1.  Does anyone know why that would be?  It's not because of the axes vector itself, which contains signed integers.

Reply:
It could be due to `normalize_attributes()`. before calling to `normalize_compute_shape()`, it first calls into `normalize_attributes()`.

Reply:
axes values are parsed negative from ONNX, but it is during evaluation time before calling `compute_shape()` for slice operator, MIGraphX would call `normalize_attributes()` which would convert those negative value to positive ones.